src/ImageClassifier/index.js

```diff
@@ -44,36 +54,65 @@ class ImageClassifier {
     this.signalStop = false; // Signal to stop the loop
     this.prevCall = ""; // Track previous call to detectStart() or detectStop()
 
-    if (typeof modelNameOrUrl === "string") {
-      if (MODEL_OPTIONS.includes(modelNameOrUrl)) {
-        this.modelName = modelNameOrUrl;
-        this.modelUrl = null;
-        switch (this.modelName) {
-          case "mobilenet":
-            this.modelToUse = mobilenet;
-            this.version = options.version || DEFAULTS.mobilenet.version;
-            this.alpha = options.alpha || DEFAULTS.mobilenet.alpha;
-            this.topk = options.topk || DEFAULTS.mobilenet.topk;
-            break;
-          case "darknet":
-            this.version = "reference"; // this a 28mb model
-            this.modelToUse = darknet;
-            break;
-          case "darknet-tiny":
-            this.version = "tiny"; // this a 4mb model
-            this.modelToUse = darknet;
-            break;
-          case "doodlenet":
-            this.modelToUse = doodlenet;
-            break;
-          default:
-            this.modelToUse = null;
-        }
-      } else {
-        // its a url, we expect to find model.json
-        this.modelUrl = modelNameOrUrl;
-        // The teachablemachine urls end with a slash, so add model.json to complete the full path
-        if (this.modelUrl.endsWith("/")) this.modelUrl += "model.json";
+    if (typeof modelNameOrUrl === "string" && isHttpUrl(modelNameOrUrl)) {
```

Comment:
What if it's a relative path?  ie `ml5.imageClassifier("/model/model.json")`

Reply:
Yeah, that would not work. Thank you for the catch! Though I can't think of a way to cleanly handle this using the `handleModelName` function.